nucypher/blockchain/eth/token.py

```diff
@@ -501,6 +508,9 @@ def start(self, force: bool = False) -> None:
         d.addErrback(self.handle_tracking_errors)
         self.log.info(f"STARTED PERIOD TRACKING with {len(self.__actions)} registered actions.")
 
+        if act_now:
```

Comment:
To be fair, instead of `act_now` + tracking how many times we confirmed I'd much rather read on chain if we need a confirmation periodically (and if we do, confirm it). What do you think?

Reply:
> To be fair, instead of act_now + tracking how many times we confirmed...

I feel confused.  Unless I am missing something - I dont think we are tracking "how many times we confirmed" anywhere.  

>  I'd much rather read on chain if we need a confirmation periodically ...

This is very close to what we are already doing.  We check once an hour if the *on-chain* period has changed, and if it has, then we check if a confirmation is needed.

Reply:
Hmm, ok, I think I've missed it, or the diff didn't show...

Reply:
:heavy_check_mark: - Made updates for failure tolerance.